### PR TITLE
Make singleton tasks less likely to collide with the 'current' one when requeuing from a task

### PR DESCRIPTION
**Pull request by iniq**
The logic described in the comment didn't match the conditions being generated - basically a non-errored, but fetched, task would count as one to not duplicate. This is one in progress - which should not be matched against.

## Testing Notes
 * easiest way to test is to try to use the singleton call to queue a new task with the same data pattern *from an instance of this task*.
 * you can very easily do this off of the linked pull, that required this change: https://github.com/TribeHR/TribeHR/pull/2859

### CODE REVIEW

**Comment by phileverson** [comment]
:fork_and_knife:

**Comment by phileverson** [comment]
Tested with https://github.com/TribeHR/TribeHR/pull/2859 - works as expected. 
:+1: 
:100:

File: Model/QueuedTask.php [hunk 1]
```diff
@@ -72,7 +72,7 @@ public function createSingletonJob($jobName, $data, $notBefore = null, $group =
 				'completed' => null,
 				'or' => array(
 					'fetched' => null,
-					'failed' => 0,
+					'failed >' => 0,
```
Comment (KoryHodgson) [inline Model/QueuedTask.php]: I think this should be failed < 5  otherwise if a task ever failed out it won't ever be recreated

Comment (iniq) [inline Model/QueuedTask.php]: I'll update.
But the logic does need to change - we were getting collisions against a currently-in-progress instance, which is bad. We only want to match against 'pending' ones.

**Comment by KoryHodgson** [comment]
:+1:

**Comment by phileverson** [comment]
Discussed in person, re-tested, working as expected. 
:+1: :100:

**Comment by iniq** [comment]
Merging for @phileverson

**Pull request merged**